src/app/state/hooks.js

```diff
@@ -149,15 +168,76 @@ export function useCoordination(parameters, coordinationScopes) {
     }));
   }, shallow);
 
-  const setters = fromEntries(parameters.map((parameter) => {
+  const setters = useMemo(() => fromEntries(parameters.map((parameter) => {
     const setterName = `set${capitalize(parameter)}`;
     const setterFunc = value => setCoordinationValue({
       parameter,
       scope: coordinationScopes[parameter],
       value,
     });
     return [setterName, setterFunc];
-  }));
+  })), [parameters, coordinationScopes]);
+
+  return [values, setters];
+}
+
+const coordinationScopesMap = {
```

Comment:
Can you add a comment about how this map works for future developers / ourselves

Reply:
```suggestion
const AUXILIARY_COORDINATION_TYPES_MAP = {
```

Reply:
(This is a map from normal coordination types to auxiliary coordination types, not scopes)